src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -711,18 +711,28 @@ void grpc_chttp2_encode_header(grpc_chttp2_hpack_compressor* c,
   }
   for (size_t i = 0; i < extra_headers_size; ++i) {
     grpc_mdelem md = *extra_headers[i];
-    uintptr_t static_index = grpc_chttp2_get_static_hpack_table_index(md);
-    if (static_index) {
-      emit_indexed(c, static_cast<uint32_t>(static_index), &st);
+    const bool is_static =
+        GRPC_MDELEM_STORAGE(md) == GRPC_MDELEM_STORAGE_STATIC;
+    uintptr_t static_index;
+    if (is_static &&
```

Comment:
would the comparison with GRPC_CHTTP2_LAST_STATIC_ENTRY always be true?

Reply:
No, because there H2 only uses a subset of the static metadata in core. GRPC_CHTTP2_LAST_STATIC_ENTRY is 61, while there's something like 80+ in core.